python/lib/communication/dmod/communication/message.py

```diff
@@ -211,83 +151,47 @@ def get_response_to_type(cls) -> Type[AbstractInitRequest]:
         """
         return cls.response_to_type
 
-    def __init__(self, data=None, *args, **kwargs):
-        super(Response, self).__init__(*args, **kwargs)
-        self.data = data
-
-    def __eq__(self, other):
-        return self.success == other.success and self.reason == other.reason and self.message == other.message \
-               and self.data == other.data
-
-    def to_dict(self) -> dict:
-        serial = super(Response, self).to_dict()
-        if self.data is None:
-            serial['data'] = {}
-        elif isinstance(self.data, dict):
-            serial['data'] = self.data
-        else:
-            serial['data'] = self.data.to_dict()
-        return serial
-
 
 class InvalidMessage(AbstractInitRequest):
     """
     An implementation of :class:`Message` to model deserialized request messages that are not some other valid message
     type.
     """
 
-    event_type: MessageEventType = MessageEventType.INVALID
+    event_type: ClassVar[MessageEventType] = MessageEventType.INVALID
     """ :class:`MessageEventType`: the type of ``MessageEventType`` for which this message is applicable. """
 
-    @classmethod
-    def factory_init_from_deserialized_json(cls, json_obj: dict):
-        """
-        Factory create a new instance of this type based on a JSON object dictionary deserialized from received JSON.
-
-        Parameters
-        ----------
-        json_obj
-
-        Returns
-        -------
-        A new object of this type instantiated from the deserialize JSON object dictionary, or none if the provided
-        parameter could not be used to instantiated a new object.
-        """
-        try:
-            return cls(content=json_obj['content'])
-        except:
-            return None
-
-    def __init__(self, content: dict):
-        self.content = content
-
-    def to_dict(self) -> dict:
-        return {'content': self.content}
+    content: Dict[str, Any]
 
 
 class InvalidMessageResponse(Response):
 
-    response_to_type = InvalidMessage
+    response_to_type: ClassVar[Type[AbstractInitRequest]] = InvalidMessage
     """ The type of :class:`AbstractInitRequest` for which this type is the response"""
 
-    def __init__(self, data=None):
-        super().__init__(success=False,
-                         reason='Invalid Request Message',
-                         message='Request message was not formatted as any known valid type',
-                         data=data)
+    success = False
+    reason: Literal["Invalid Request message"] = "Invalid Request message"
+    message: Literal["Request message was not formatted as any known valid type"] = "Request message was not formatted as any known valid type"
+    data: Optional[Serializable]
 
+    def __init__(self, data: Optional[Serializable]=None, **kwargs):
+        super().__init__(data=data)
+
+
+class HttpCode(Serializable):
+    http_code: int = Field(ge=100, le=599)
```

Comment:
Maybe? This seems like it may depend on answers to your below questions.

Reply:
Having a default of `500` for the HTTP code within _ErrorResponse_ seems to make sense, and I'll defer to @christophertubbs since he created that type.  Having a default of `500` on _HttpCode_ doesn't make sense to me:  we should consider removing the type if it's being used so narrowly for a default to seem appropriate.

Reply:
The default for `HttpCode` to `500` only seemed to make sense based on how I saw it used. I figured that if that's the majority use case, rock on with that. Otherwise, big shrug, it's doing what we want.